crates/apub/src/http/mod.rs

```diff
@@ -44,13 +44,14 @@ pub async fn shared_inbox(
 ) -> Result<HttpResponse, LemmyError> {
   let unparsed = payload_to_string(payload).await?;
   info!("Received shared inbox activity {}", unparsed);
+  let activity_data: ActivityCommonFields = serde_json::from_str(&unparsed)?;
   let activity = serde_json::from_str::<WithContext<SharedInboxActivities>>(&unparsed)?;
   match activity.inner() {
     SharedInboxActivities::GroupInboxActivities(g) => {
-      receive_group_inbox(g, request, &context).await
+      receive_group_inbox(g, activity_data, request, &context).await
     }
     SharedInboxActivities::PersonInboxActivities(p) => {
-      receive_person_inbox(p, request, &context).await
+      receive_person_inbox(p, activity_data, request, &context).await
```

Comment:
Also not that elegant here as we have to pass this extra var around, but maybe this can be rewritten better.

Reply:
Instead of the two separate deserializations, we could also do a single one into `WithContext<Activity<SharedInboxActivities>>`, where Activity would have the id and actor fields which are now in `ActivityCommonFields`. 

But there are a few problems with that:
- actor is always an url, and cant be ObjectId<Person> or ObjectId<Community>
- need to rewrite all the activity handlers to pass in id and actor somehow, as they wont be in self anymore

Reply:
Okay i got a different idea.